### PR TITLE
Adding longer location name transformation

### PR DESCRIPTION
**Pull request by ashish3586**
will generate transformation like.
(Michael Jordan is a professor at Berkeley ., B-PER I-PER O O O O B-LOC O) ->
 (Michael Jordan is a professor at Eastern Berkeley ., B-PER I-PER O O O O B-LOC I-LOC O)

### CODE REVIEW

File: transformations/longer_location_ner/test.json [hunk 1]
```diff
@@ -0,0 +1,51 @@
+{
+  "type": "longer_location_ner",
+  "test_cases": [
+    {"class": "LongerLocationNer",
+      "inputs": {
+        "token_sequence": "I am travelling to London .",
+        "tag_sequence": "O O O O B-LOC O"
+      },
+      "outputs": {
```
Comment (aadesh11) [inline transformations/longer_location_ner/test.json:9]: The output should be a list of dict.

File: transformations/longer_location_ner/transformation.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        token_seq.insert(b_index+1, word)
+        tag_seq.insert(b_index+1, i_tag)
+    return token_seq, tag_seq
+
+
+class LongerLocationNer(TaggingOperation):
+    tasks = [TaskType.TEXT_TAGGING]
+    languages = "All"
+
+    def __init__(self):
```
Comment (aadesh11) [inline transformations/longer_location_ner/transformation.py:40]: Please take max_output as one of the arguments, which decides the max number of outputs the transformation will return.

File: transformations/longer_location_ner/transformation.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        i_tag = "I-" + tag
+        assert len(token_sequence) == len(tag_sequence), \
+            "Lengths of `token_sequence` and `tag_sequence` should be the same"
+        if b_tag in tag_sequence:
+            b_tag_index = tag_sequence.index(b_tag)
+            i_tag_index = b_tag_index + 1
+            token_sequence, tag_sequence = create_token_and_tag(b_tag_index, i_tag_index, b_tag, i_tag
+                                                                , token_sequence, tag_sequence)
+        assert len(token_sequence) == len(tag_sequence)
+        return token_sequence, tag_sequence
```
Comment (aadesh11) [inline transformations/longer_location_ner/transformation.py:59]: return list of a tuple of token and tag sequences.

**Comment by kaustubhdhole** [comment]
@ashish3586 please fix the PR so that the build can pass.

File: transformations/longer_location_ner/test.json (hunk 1 above)
Comment (ashish3586) [reply transformations/longer_location_ner/test.json]: done

File: transformations/longer_location_ner/transformation.py (hunk 3 above)
Comment (ashish3586) [reply transformations/longer_location_ner/transformation.py]: done

File: transformations/longer_location_ner/transformation.py (hunk 2 above)
Comment (ashish3586) [reply transformations/longer_location_ner/transformation.py]: done

File: transformations/longer_location_ner/transformation.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+"""
+A tagging implementation of NER systems.
+I am travelling to London. --> I am travelling to South/Central London.
+"""
+
+
+def pick_random_word(seed):
+    # pick random "pre" or "post" perturb type with respective word
+    random.seed(seed)
+    pre_word_list = ["East", "West", "South", "North", "Eastern", "Western", "Central"]
```
Comment (kaustubhdhole) [inline transformations/longer_location_ner/transformation.py:15]: Berkley 
--> Southern Berkley 
--> Northern Berkley
--> Northeast Berkley
--> Southwest Berkley etc.
--> South West Berkley (3 tags could be added here)
--> SE Berkley 

Secondary inter-cardinal directions (check here: https://en.wikipedia.org/wiki/Cardinal_direction)
(Northern North East or NNE I think and those like that)

**Comment by kaustubhdhole** [comment]
@ashish3586 would you like to address the reviewer comments.

**Comment by ashish3586** [comment]
> @ashish3586 would you like to address the reviewer comments.

Changes are ready. will push shortly. Thanks for reminding me.

**Comment by ashish3586** [comment]
> > @ashish3586 would you like to address the reviewer comments.
> 
> Changes are ready. will push shortly. Thanks for reminding me.

File: transformations/longer_location_ner/transformation.py (hunk 4 above)
Comment (ashish3586) [reply transformations/longer_location_ner/transformation.py]: completed all suggested changes.

**Review by kaustubhdhole — APPROVED** [review]

**Pull request merged**